worlds/wl4/items.py

```diff
@@ -0,0 +1,229 @@
+    'Golden Fish':                      ItemData(ItemType.TREASURE, 0x72,                              IC.progression_skip_balancing),
+    'Golden Candle Holder':             ItemData(ItemType.TREASURE, 0x73,                              IC.progression_skip_balancing),
+    'Golden Lamp':                      ItemData(ItemType.TREASURE, 0x74,                              IC.progression_skip_balancing),
+    'Golden Crescent Moon Bed':         ItemData(ItemType.TREASURE, 0x75,                              IC.progression_skip_balancing),
+    'Golden Teddy Bear':                ItemData(ItemType.TREASURE, 0x76,                              IC.progression_skip_balancing),
+    'Golden Lollipop':                  ItemData(ItemType.TREASURE, 0x77,                              IC.progression_skip_balancing),
+    'Golden Game Boy Advance':          ItemData(ItemType.TREASURE, 0x78,                              IC.progression_skip_balancing),
+    'Golden Robot':                     ItemData(ItemType.TREASURE, 0x79,                              IC.progression_skip_balancing),
+    'Golden Rocket':                    ItemData(ItemType.TREASURE, 0x7A,                              IC.progression_skip_balancing),
+    'Golden Rocking Horse':             ItemData(ItemType.TREASURE, 0x7B,                              IC.progression_skip_balancing),
+    'Full Health Item':                 ItemData(ItemType.ITEM,     0x80,                              IC.useful),
+    'Wario Form Trap':                  ItemData(ItemType.ITEM,     0x81,                              IC.trap),
+    'Heart':                            ItemData(ItemType.ITEM,     0x82,                              IC.filler),
+    'Lightning Trap':                   ItemData(ItemType.ITEM,     0x83,                              IC.trap),
+    'Minigame Coin':                    ItemData(ItemType.ITEM,     0x84,                              IC.filler),
+#   'Entry Passage Clear':              ItemData(ItemType.EVENT,    None,                              IC.filler),
+    'Emerald Passage Clear':            ItemData(ItemType.EVENT,    None,                              IC.progression),
+    'Ruby Passage Clear':               ItemData(ItemType.EVENT,    None,                              IC.progression),
+    'Topaz Passage Clear':              ItemData(ItemType.EVENT,    None,                              IC.progression),
+    'Sapphire Passage Clear':           ItemData(ItemType.EVENT,    None,                              IC.progression),
+    'Escape the Pyramid':               ItemData(ItemType.EVENT,    None,                              IC.progression),
+}
+
+
+def filter_items(*, type: ItemType = None, passage: Passage = None) -> Iterable[Tuple[str, ItemData]]:
+    items = item_table.items()
+    if type != None:
+        items = filter(lambda i: i[1].type == type, items)
+    if passage != None:
+        items = filter(lambda i: i[1].passage() == passage, items)
```

Comment:
In this case I don't really care what the actual types are because this function is typed `Iterable`, so constructing lists feels pointless here. If type checking is the issue, I can add that hint to `items` instead.

Reply:
Marking `items` as an Iterable should work as well